pkg/version/version.go

```diff
@@ -34,26 +30,31 @@ type CiliumVersion struct {
 	GoRuntimeVersion string
 	// Arch is the architecture where Cilium was compiled
 	Arch string
+	// AuthorDate is the git author time reference stored as string ISO 8601 formatted
+	AuthorDate string
 }
 
+// Version is set during build
 var Version string
 
-func versionFrom(versionString string) CiliumVersion {
-	cver := CiliumVersion{}
-	output := strings.Replace(versionString, "Cilium ", "", 1)
-
-	fields := strings.Split(output, " ")
-	cver.Version = fields[0]
-	cver.Revision = fields[1]
-	cver.Arch = fields[len(fields)-1]
-	cver.GoRuntimeVersion = runtime.Version()
+// FromString converts a version string into struct
+func FromString(versionString string) CiliumVersion {
+	// string to parse: "0.13.90 a722bdb 2018-01-09T22:32:37+01:00 go version go1.9 linux/amd64"
+	fields := strings.Split(versionString, " ")
 
+	cver := CiliumVersion{
```

Comment:
Can you check for `len(fields)` to avoid panics?

Reply:
Fixed!